New package: PoGOGamemaster v1.0.0

JuliaRegistrator opened the pull request:
- Registering package: PoGOGamemaster
- Repository: https://github.com/JuliaPokemonGO/PoGOBase.jl
- Created by: @drnobody42
- Version: v1.0.0
- Commit: 19e5b3024988d6c6bbbe9e5c11ecc1d6cc4e7d4e
- Reviewed by: @drnobody42
- Reference: https://github.com/JuliaPokemonGO/PoGOBase.jl/commit/19e5b3024988d6c6bbbe9e5c11ecc1d6cc4e7d4e#commitcomment-154229563
- Description: Base package for Pokemon GO in Julia
<!-- e439c565d9fc1aafd116a836874b939539b7cf71e7c6e9f326fd1ae9425e36c0b73e305b2659c878292aaaecd131dcbccbfa1a4741dd867e93b192e82157e7906249bf9a2c73a147a66b60de26e865e898708105b7dca3d7b3c4ffddd16ba29f59e90aaf1f1ee39d60d710b0cd2d6dcd32123d03a67c63ba9591a7606b752243d32171bc0aaafa3eacee3344c40b7cc9dfb3de4015ba82cbe678b133d7dca46fcb3b812ae6d098c8bde7804c859addabb9a365ce950deb1b9a1974712f466fd3698511fd997e819334da23749d2fe31bc9ebdc98d0b0c56144c55be34d4834dc189c42afd7f57fa62735e8fe2cc62f5e -->

goerz commented:
I'm a little confused by the organization of this repository. It seems like a package `PoGOGamemaster` might be intended as a user-facing package, but `PoGOBase` would then seem like an internal package. So it seems unusual to have `PoGOGamemaster` as a subdir-package of `PoGOBase`, and not the other way around. Is this what you intend? If so, it might be best to explain the structure of the project somewhere.

In any case, the links to the documentation in the README are currently broken, so there isn't really any documentation of the package at this time. A minimum amount of documentation is generally a [requirement for registration](https://discourse.julialang.org/t/when-does-it-make-sense-to-post-a-new-package-to-the-julia-repository/108701/4).

For a version `v1.0` especially, this is even more important. If you are following [semantic versioning](https://semver.org) you're saying that the package already has a stable API. That requires specifying what that API is, i.e., a complete API documentation. I would also recommend making sure that you have a full test suite with good coverage for a version 1.0. Otherwise, it is rather difficult to identify breaking or non-breaking changes, going forward, as semantic versioning requires.

I don't usually recommend that people register new packages at `v1.0`. If your package can live up to the expectations of a "stable" package, then by all means. Otherwise, it might be better to register the initial public version as `v0.1`. Once people have had a chance to interact with the package, so that you have an idea about whether the API is okay as it is, you can tag a `v1.0` release at any point when the documentation is sufficiently polished.

GunnarFarneback commented:
[noblock]
> It seems like a package PoGOGamemaster might be intended as a user-facing package, but PoGOBase would then seem like an internal package. So it seems unusual to have PoGOGamemaster as a subdir-package of PoGOBase, and not the other way around.

For what it's worth, if you have multiple packages in the same repository a better organization is to have all of them in separate subdirectories. That way you don't get unused copies of the "subdir" packages distributed with the "main" package.

drnobody42 commented:
[noblock] PoGOGamemaster is internal and not intended for users. PoGOBase is the user-facing package. During `deps/build.jl`, only PoGOGamemaster gets loaded, it then parses `gamemaster.json`, and the results are (re)serialized in Julia form as `deps/gamemaster.ser`. This is what gets loaded when you build PoGOBase.

goerz commented:
> PoGOBase is the user-facing package

That's unusual… But I'll have to take your word for it. Maybe I'm just misunderstanding the name. Does "Base" here mean something different from "Base" in, e.g., [`SciMLBase`](https://github.com/SciML/SciMLBase.jl)? That is, possibly "Base" means something specific in Pokemon Go, and this isn't a base package for a larger ecosystem, the way "Base" is commonly used as part of Julia package names. 

> Since `PoGOBase` depends on `PoGOGamemaster`, the docs won't build until this gets registered.

Technically, since it's a monorepo, you could (and maybe should) dev-install `PoGOGamemaster` in `docs/make.jl`. But, fair enough.

Sorry, I always get a bit confused with monorepos. So I was looking at the README and linked docs for `PoGOBase`. As for [`PoGOGamemaster `](https://github.com/JuliaPokemonGO/PoGOBase.jl/tree/main/PoGOGamemaster), I still recommend having an independent README file that explains the role of `PoGOGamemaster` and its relation to the main `PoGOBase`. It's just a good idea to have a README for _every_ registered package. But it's up to you.

Once you get around to registering `PoGOBase`, you'd definitely have to extend the README and/or documentation for that package.

> It's been in private usage for about 2 years.

I just wanted to make sure you're aware of the expectations that a v1.0 version carries, with semantic versioning. Ultimately, this is between you and your users.

One small additional note: Someone internally raised some concerns that this package might involve "re-engineering" and may violate the [terms of service](https://nianticlabs.com/terms) of the company owning Pokemon GO. We're in no position to judge potential TOS or trademark violations in the General registry. So, that's something for you to determine. You'll probably want to make sure that you don't open yourself up to any kind of legal trouble.

[noblock]

JuliaRegistrator committed:
New package: PoGOGamemaster v1.0.0

UUID: 62c09a92-b096-4650-ba41-ec5e38cc10f2
Repo: https://github.com/JuliaPokemonGO/PoGOBase.jl.git
Tree: 4bbf272bd769a357f246facecb8804c406984c08

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a
